### ISSUE TITLE
Product description page is missing

### ISSUE THREAD

**Issue by Nitish-hack** [MilanCommunity/Milan#701]
### Description

there should be a products description page i.e. when user clicks on a product a description page of that product should open 

### 🥦 Browser

Google Chrome

### Checklist ✅

- [X] I checked and didn't find similar issue
- [X] I have read the [Contributing Guidelines](https://github.com/IAmTamal/Milan/blob/main/CONTRIBUTING.md)
- [X] I am willing to work on this issue (blank for no)

**Comment by Nitish-hack** [comment]
I would like to contribute to this issue as a part of @GSSOC'23
Please assign it to me. Thank you.

**Comment by IAmTamal** [comment]
This will be discussed once we are done with #700

**Comment by tamalCodes** [comment]
Hi @Nitish-hack , you can work on this !

**Comment by tamalCodes** [comment]
Hi there, so TBH I would love to have both, since without an API that actually fetches the product and gives you stuff, a Frontend makes no sense. So how about in this issue we make an API, then once it is done we can make another one to properly and safely integrate the Frontend to the API ? 

Also about the fields, I think we can show the 
 - productType
 - productName
 - productPrice
 - productDescription
 - productImage
 - productQty

And along with that the route I think can be `/shop/[productSlug]`

**Comment by Nitish-hack** [comment]
yeah, we should first design an api to fetch single product from database either by id or product slug passed in the request? so should I make another issue regarding this?

**Comment by tamalCodes** [comment]
Sounds good, so that would be 2 issues in total.

 - Fetch a single product from the DB and return the details to the FE
 - Integrate the backend to the FE, to display details about a product.

**Comment by Nitish-hack** [comment]
so i have created an issue for api to fetch single product  ( issue : #804)
please assign that to me  so that i can start working , then we will work on integrating it and designing product description page

**Comment by tamalCodes** [comment]
Sounds good. This will worked upon later then.

**Comment by Nitish-hack** [comment]
yes of course

**Comment by Nitish-hack** [comment]
Hi @tamalCodes  i have created the api for single product and its merged too ......now please assign this to me so that i can work further

**Comment by tamalCodes** [comment]
Cool, you can start to work on this !

**Comment by Nitish-hack** [comment]
okay great

**Comment by Sa91001** [comment]
Hi there, it has been a while. I would love to know what changes you are making and/or are working on this. Please reply shortly.

**Comment by Nitish-hack** [comment]
actually my exams are going on, so I am not able to work on it..... if its needed urgently you can assign it to someone else

**Comment by JatinBhargava** [comment]
Hi, @tamalCodes I have a request can I work on this issue? I am fairly new to open source so if you give me a chance I can try. Also, how can I contact you and whoever is in charge of these issues?

**Comment by tamalCodes** [comment]
Hi @JatinBhargava , you can start working on it. The person who has been assigned has exams - and of course, exams come first. You can start working on this.

**Comment by JatinBhargava** [comment]
Hi @tamalCodes, I want to ask do we have some UI/UX for a single product page. I am able to get a new page for a product when a user clicks on the product's name. Right now I am doing in React Only. 
2) what should be the URL?
right now it is /shop/:category/:slug

Next Up: need to integrate with the backend. 

![image](https://github.com/tamalCodes/Milan/assets/63741896/10868b73-8179-45f1-891a-a5465e84a2c8)

Note: No issue, whenever the person is assigned to this task, come, will take over. I just want to try it out as this is my first time working on open source project.

**Comment by JatinBhargava** [comment]
Hi @tamalCodes, I am able to get the data from database to front end. Only the issue is not getting image. But for that We need to make some changes in post request (need to add a middleware : multer). So may I do change the existing route or make my own with the middleware addition. 

Note : Should I make a pull request so that you can see the changes? 

Here's the quick peek what I am getting. 
![image](https://github.com/tamalCodes/Milan/assets/63741896/26d204a1-14c9-44d9-b1fa-651559a22fd8)

**Comment by tamalCodes** [comment]
Hi @JatinBhargava , the thing us we donot have any image at place, It is static, so it won't come in background.
For now you can skip that part, make sure to have an image field as `string` but you can skip it's functionality.

As of now we can proceed with it being totally static.
